contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,876 @@
+
+    IBSCValidatorSet.Validator[] private _eligibleValidators;
+    bytes[] private _eligibleValidatorVoteAddrs;
+
+    struct Validator {
+        address consensusAddress;
+        address operatorAddress;
+        address poolModule;
+        bytes voteAddress;
+        Description description;
+        Commission commission;
+        bool jailed;
+        uint256 jailUntil;
+        uint256 updateTime;
+        uint256[20] __reservedSlots;
+    }
+
+    struct Description {
+        string moniker;
+        string identity;
+        string website;
+        string details;
+    }
+
+    struct Commission {
+        uint64 rate; // the commission rate charged to delegators(10000 is 100%)
+        uint64 maxRate; // maximum commission rate which validator can ever charge
+        uint64 maxChangeRate; // maximum daily increase of the validator commission
+    }
+
+    struct SlashRecord {
+        uint256 jailUntil;
+        uint256 slashAmount;
+        uint248 slashHeight;
+        SlashType slashType;
+    }
+
+    enum SlashType {
+        DoubleSign,
+        DownTime,
+        MaliciousVote
+    }
+
+    /*----------------- events -----------------*/
+    event ValidatorCreated(
+        address indexed consensusAddress, address indexed operatorAddress, address indexed poolModule, bytes voteAddress
+    );
+    event ConsensusAddressEdited(
+        address indexed operatorAddress, address indexed oldAddress, address indexed newAddress
+    );
+    event CommissionRateEdited(address indexed operatorAddress, uint64 commissionRate);
+    event DescriptionEdited(address indexed operatorAddress);
+    event VoteAddressEdited(address indexed operatorAddress, bytes newVoteAddress);
+    event Delegated(address indexed operatorAddress, address indexed delegator, uint256 shares, uint256 bnbAmount);
+    event Undelegated(address indexed operatorAddress, address indexed delegator, uint256 shares, uint256 bnbAmount);
+    event Redelegated(
+        address indexed srcValidator,
+        address indexed dstValidator,
+        address indexed delegator,
+        uint256 oldShares,
+        uint256 newShares,
+        uint256 bnbAmount
+    );
+    event RewardDistributed(address indexed operatorAddress, uint256 reward);
+    event RewardDistributeFailed(address indexed operatorAddress, bytes failReason);
+    event ValidatorSlashed(
+        address indexed operatorAddress,
+        uint256 jailUntil,
+        uint256 slashAmount,
+        uint248 slashHeight,
+        SlashType slashType
+    );
+    event ValidatorJailed(address indexed operatorAddress);
+    event ValidatorEmptyJailed(address indexed operatorAddress);
+    event ValidatorUnjailed(address indexed operatorAddress);
+    event Claimed(address indexed operatorAddress, address indexed delegator, uint256 bnbAmount);
+    event StakingPaused();
+    event StakingResumed();
+
+    /*----------------- modifiers -----------------*/
+    modifier onlyInitialized() {
```

Comment:
I think `onlyInitialized` is no longer needed.

Because the consensus engine will do the init work after it is deployed.

Reply:
Done